File: frontend/src/utils/netDebt.ts
/** Чистый долг = Долг − Наличность (млн валюты отчёта). */
export function computeNetDebt(
  debt: number | null | undefined,
  cashAndEquivalents: number | null | undefined,
): number | null {
  if (debt == null || cashAndEquivalents == null) return null;
  return Math.round((debt - cashAndEquivalents) * 1000) / 1000;
}

File: backend/app/services/analysis/net_debt.py
"""Чистый долг (Net Debt) = Долг − Наличность, млн валюты отчёта."""
from typing import Optional


def compute_net_debt(
    debt: Optional[float],
    cash_and_equivalents: Optional[float],
) -> Optional[float]:
    if debt is None or cash_and_equivalents is None:
        return None
    return round(float(debt) - float(cash_and_equivalents), 3)
